### PR TITLE
Add the ability for admins to view accounts pending creation

### PR DESCRIPTION
**Pull request by kfkitsune**
Provides a new admin-restricted page to view, and optionally purge, account creation requests in the ``logincreate`` table. In the event of an invalid ``logincreate`` record, the email wouldn't normally be displayed. So this also adds an ``invalid_email_addr`` field to the ``logincreate`` table to permit storing the originally provided email (since the email field is unique to the table).

Addresses the first and second proposals in #302.

### CODE REVIEW

File: weasyl/controllers/admin.py [hunk 1]
```diff
@@ -122,6 +122,42 @@ def admincontrol_acctverifylink_(request):
...
+def admincontrol_pending_accounts_get_(request):
+    """
+    Retrieve a listing of any active pending accounts in the logincreate table.
+
+    :param request: The Pyramid request object.
+    :return: A Pyramid response with a webpage containing the pending accounts.
+    """
+    query = d.engine.execute("""
+        SELECT token, username, email, invalid_email_addr, unixtime
+        FROM logincreate
```
Comment (charmander) [inline weasyl/controllers/admin.py]: This should probably have an order of some… sort.

File: weasyl/controllers/admin.py [hunk 2]
```diff
@@ -122,6 +122,42 @@ def admincontrol_acctverifylink_(request):
...
+@token_checked
+def admincontrol_pending_accounts_post_(request):
+    """
+    Purges a specified logincreate record.
+
+    :param request: A Pyramid request.
+    :return: HTTPSeeOther to /admincontrol/pending_accounts
+    """
+    form = request.web_input(logincreatetoken="")
+    print(form.logincreatetoken)
```
Comment (charmander) [inline weasyl/controllers/admin.py]: Leftover debugging?

File: weasyl/controllers/admin.py [hunk 3]
```diff
@@ -122,6 +122,42 @@ def admincontrol_acctverifylink_(request):
...
+@admin_only
+@token_checked
+def admincontrol_pending_accounts_post_(request):
+    """
+    Purges a specified logincreate record.
+
+    :param request: A Pyramid request.
+    :return: HTTPSeeOther to /admincontrol/pending_accounts
+    """
+    form = request.web_input(logincreatetoken="")
```
Comment (charmander) [inline weasyl/controllers/admin.py]:
```python
token = request.POST.get('logincreatetoken')
```

File: weasyl/cron.py [hunk 4]
```diff
@@ -8,7 +8,7 @@
 
 
 def run_periodic_tasks():
-    now = arrow.utcnow()
+    now = arrow.utcnow().timestamp
```
Comment (charmander) [inline weasyl/cron.py]: Instead of this, how about passing `time=now - datetime.timedelta(days=2)` to the query?

Comment (kfkitsune) [reply weasyl/cron.py]: That would still need to be converted to a numeric timestamp, it seems.

```
(tmp-hnzdl9RK) kyra@kitty ~/g/tmp> python
Python 3.5.3 (default, Jan 19 2017, 14:11:04)
[GCC 6.3.0 20170118] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> import datetime
>>> now = arrow.utcnow()
>>> now - datetime.timedelta(days=2)
<Arrow [2018-01-14T21:47:18.110156+00:00]>
>>> res = now - datetime.timedelta(days=2)
>>> type(res)
<class 'arrow.arrow.Arrow'>
>>> res.timestamp
1515966438
>>>
```

We can use ``timedelta``, if we want but we'd need turn around and issue a ``.timestamp``, otherwise we'd be right back to where we are now with the TypeError against the stored value and what we're setting as the ``time`` parameter.
```
>>> (now - datetime.timedelta(days=2)).timestamp
1515966438
```

Personally, I am fine with either method, but I am not sure if one is better than the other in this situation.

Comment (charmander) [reply weasyl/cron.py]: Oh, `.timestamp` is an int? How are `now.minute` and `now.hour` working below?

Comment (kfkitsune) [reply weasyl/cron.py]: Okay, so I had to dig in to the source for this. It seems it [behaves as if it were](https://github.com/crsmithdev/arrow/blob/master/arrow/arrow.py#L25) a ``datetime`` object, but it implements additional functionality. So the original ``datetime`` can be accessed:

```
>>> now._datetime
datetime.datetime(2018, 1, 17, 0, 27, 48, 796416, tzinfo=tzutc())
>>> now._datetime - datetime.timedelta(days=2)
datetime.datetime(2018, 1, 15, 0, 27, 48, 796416, tzinfo=tzutc())
```

But it also responds to typical ``datetime`` attributes such as ``.hour / .minute``.

```
(tmp-hnzdl9RK) kyra@kitty ~/g/tmp> python
Python 3.5.3 (default, Jan 19 2017, 14:11:04)
[GCC 6.3.0 20170118] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import datetime
>>> import arrow
>>> now = arrow.utcnow()
>>> type(now)
<class 'arrow.arrow.Arrow'>
>>> now.hour
0
>>> now.minute
27
>>> type(now.hour)
<class 'int'>
>>> type(now.minute)
<class 'int'>

>>> today = datetime.datetime.today()
>>> today
datetime.datetime(2018, 1, 16, 19, 36, 6, 186271)
>>> today.minute
36
>>> today.hour
19
>>> type(today)
<class 'datetime.datetime'>
>>> type(now._datetime)
<class 'datetime.datetime'>
```

It looks like it is getting the ``hour``/``minute`` out [via the following code](https://github.com/crsmithdev/arrow/blob/master/arrow/arrow.py#L379):
```
def __getattr__(self, name):
        if name == 'week':
            return self.isocalendar()[1]
        if name == 'quarter':
            return int((self.month-1)/self._MONTHS_PER_QUARTER) + 1
        if not name.startswith('_'):
            value = getattr(self._datetime, name, None)
            if value is not None:
                return value
        return object.__getattribute__(self, name)
```
And I can directly replicate the above by doing:
```
>>> getattr(now._datetime, 'hour', None)
0
>>> getattr(now._datetime, 'minute', None)
27
>>> getattr(now._datetime, 'second', None)
48
>>>
```

So in summary, it looks like it directly passes through the attribute if you do a ``arrow.$NAME``, when the property doesn't begin with an underscore, and gets the attribute from the underlying ``datetime`` object. Having dug at this a bit, this makes sense considering that the project page states that it is a "fully implemented, drop-in replacement for datetime" (and you can do .hour, etc. with datetime).

Comment (kfkitsune) [reply weasyl/cron.py]: (Which then means I should just do a ``now.timestamp`` down in the SQL parameter instead of changing the ``now`` as defined here. Hold on; synchronizing.)
